packages/smooth_app/lib/pages/prices/infinite_scroll_controller.dart

```diff
@@ -0,0 +1,99 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+/// A generic controller for handling infinite scrolling in lists.
+/// [T] is the type of items being displayed.
+/// [P] is the type of parameters used for pagination.
```

Comment:
I guess we should add `[R] is the type of result we get from the parameters`.
So that we don't have to dance tango with the result:
* we call a method with the [P]arameters, to get a [R]esult.
* from the result, we compute the pagination data to be displayed
* and we get the items of type [T] to be displayed